src/imitation/util/video_wrapper.py

```diff
@@ -60,19 +76,25 @@ def _reset_video_recorder(self) -> None:
                 base_path=str(self.directory / f"video.{self.episode_id:06}"),
                 metadata={"episode_id": self.episode_id},
             )
+            self.current_video_path = pathlib.Path(self.video_recorder.path)
 
-    def reset(self):
+    def reset(self, **kwargs):
+        new_obs = super().reset(**kwargs)
         self._reset_video_recorder()
         self.episode_id += 1
-        return self.env.reset()
+        return new_obs
 
     def step(self, action):
-        res = self.env.step(action)
+        obs, rew, done, info = self.env.step(action)
         self.video_recorder.capture_frame()
-        return res
+        info["video_path"] = self.current_video_path
+        return obs, rew, done, info
 
     def close(self) -> None:
         if self.video_recorder is not None:
             self.video_recorder.close()
             self.video_recorder = None
+        if self.delete_on_close:
+            for path in self.directory.glob("*.mp4"):
+                path.unlink()
```

Comment:
One more thing I noticed -- I don't think this will delete the directories created along the way? But we need to be careful in case someone points at "." and then it'd get deleted upon closing